### PR TITLE
Add search_type support

### PR DESCRIPTION
**Pull request by marshall-lee**
This one is for #148

### CODE REVIEW

**Comment by marshall-lee** [comment]
https://travis-ci.org/toptal/chewy/jobs/55774252 fails but because of improper ordering of database records.

File: lib/chewy/query.rb [hunk 1]
```diff
@@ -818,6 +818,20 @@ def types! *params
...
+    #
+    #   scope = UsersIndex.search_type(:count)
+    #   scope.count == 0  # no records actually fetched
+    #   scope.total == 10 # but we know a total count of them
+    #
+    #   scope = UsersIndex.aggs(max_age: { max: { field: 'age' } }).search_type(:count)
+    #   max_age = scope.aggs['max_age']['value']
+    #
+    def search_type val
+      chain { options.merge!(search_type: val) }
```
Comment (pyromaniac) [inline lib/chewy/query.rb]: Could you please use `update_options` method here?

Comment (marshall-lee) [inline lib/chewy/query.rb]: @pyromaniac okay but before it please look at this one: #150, i don't want this build to fail :smiley:

Comment (marshall-lee) [inline lib/chewy/query.rb]: Are you sure about `criteria.update_options`? I think that `Criteria` is just responsible for constructing a `{ body: ... }` part but `search_type` is not a part of body. It's just a query option just like `type` or `index`.
Or you're suggesting to add a new `update_options` method to `Query` ?

Comment (pyromaniac) [inline lib/chewy/query.rb]: https://github.com/toptal/chewy/blob/master/lib/chewy/query/criteria.rb#L43
https://github.com/toptal/chewy/blob/master/lib/chewy/query.rb#L143

This is just more cmmon way to do the same thing

Comment (marshall-lee) [inline lib/chewy/query.rb]: It's not the same thing! `criteria.update_options` updates criteria's options, not query's options.

Comment (marshall-lee) [inline lib/chewy/query.rb]: It's not possible to just `chain { criteria.update_options search_type: val }`. It requires me to move `search_type` logic to `Criteria` class. I don't think that search_type is a Criteria's responsibility. It is `Query`'s responsibility.

Comment (pyromaniac) [inline lib/chewy/query.rb]: Oh, got it now. Not sure if I'm completely agree with you, but currently it is ok.

**Pull request merged**